src/type/definition.js

```diff
@@ -102,6 +102,14 @@ export function isCompositeType(type: ?GraphQLType): boolean {
   );
 }
 
+export function typeHasFields(type: ?GraphQLType): boolean {
+  return (
+    type instanceof GraphQLObjectType ||
+    type instanceof GraphQLInterfaceType ||
+    type instanceof GraphQLInputObjectType
```

Comment:
This is a change in behavior; `GraphQLInputObjectType` shouldn't have `fields`, it should have `inputFields`, right?